pkg/operator/controller/gatewayclass/controller_test.go

```diff
@@ -0,0 +1,222 @@
+				Channel:                channel,
+				StartingCSV:            version,
+				InstallPlanApproval:    "Manual",
+			},
+		}
+	}
+
+	istio := func(version string) *sailv1.Istio {
+		return &sailv1.Istio{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "openshift-gateway",
+			},
+			Spec: sailv1.IstioSpec{
+				UpdateStrategy: &sailv1.IstioUpdateStrategy{
+					Type: "InPlace",
+				},
+				Namespace: "openshift-ingress",
+				Version:   version,
+			},
+		}
+	}
+
+	tests := []struct {
+		name            string
+		request         reconcile.Request
+		existingObjects []runtime.Object
+		expectCreate    []client.Object
+		expectUpdate    []client.Object
+		expectDelete    []client.Object
+		expectError     string
+	}{
+		{
+			name:            "Nonexistent gatewayclass",
+			request:         req("openshift-default"),
+			existingObjects: []runtime.Object{},
+			expectCreate:    []client.Object{},
+			expectUpdate:    []client.Object{},
+			expectDelete:    []client.Object{},
+			expectError:     `"openshift-default" not found`,
+		},
+		{
+			name:    "Minimal gatewayclass",
+			request: req("openshift-default"),
+			existingObjects: []runtime.Object{
+				&gatewayapiv1.GatewayClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "openshift-default",
+					},
+					Spec: gatewayapiv1.GatewayClassSpec{
+						ControllerName: gatewayapiv1.GatewayController("openshift.io/gateway-controller/v1"),
+					},
+				},
+			},
+			expectCreate: []client.Object{
+				subscription("redhat-operators", "stable", "servicemeshoperator3.v3.0.1"),
+				istio("v1.24.4"),
+			},
+			expectUpdate: []client.Object{},
+			expectDelete: []client.Object{},
+		},
+		{
+			name:    "Gatewayclass with Istio version override",
+			request: req("openshift-default"),
+			existingObjects: []runtime.Object{
+				&gatewayapiv1.GatewayClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"unsupported.do-not-use.openshift.io/istio-version": "v1.24-latest",
+						},
+						Name: "openshift-default",
+					},
+					Spec: gatewayapiv1.GatewayClassSpec{
+						ControllerName: gatewayapiv1.GatewayController("openshift.io/gateway-controller/v1"),
+					},
+				},
+			},
+			expectCreate: []client.Object{
+				subscription("redhat-operators", "stable", "servicemeshoperator3.v3.0.1"),
+				istio("v1.24-latest"),
+			},
+			expectUpdate: []client.Object{},
+			expectDelete: []client.Object{},
+		},
+		{
+			name:    "Gatewayclass with OSSM and Istio overrides",
+			request: req("openshift-default"),
+			existingObjects: []runtime.Object{
+				&gatewayapiv1.GatewayClass{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							"unsupported.do-not-use.openshift.io/ossm-catalog":  "foo",
+							"unsupported.do-not-use.openshift.io/ossm-channel":  "bar",
+							"unsupported.do-not-use.openshift.io/ossm-version":  "baz",
+							"unsupported.do-not-use.openshift.io/istio-version": "quux",
```

Comment:
Did you intent to showcase that it's possible to set any values for those annotations? Or it was just to gain some time? If it's latter, may be it would make sense to set them to values which look more real.

Reply:
I could come up with something for the OSSM version and Istio version, but I don't know what would be a realistic value for the catalog or channel other than the default values, and I didn't think it mattered for the test case.  What would be some more real looking values?  Do you just want an example that has a similar length and uses similar characters to the default values?

Reply:
I thought more of giving bad examples of how the annotation values can look like. It's our testing which doesn't oblige us to anything.

> I don't know what would be a realistic value for the catalog or channel other than the default values

Channel can be `stable-3.0`, catalog can be any catalog source from the cluster something like `community-operators`?